gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ProjectLabelIT.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.gradle.plugin.tests;
+
+import net.minidev.json.parser.ParseException;
+import org.eclipse.jkube.kit.common.ResourceVerify;
+import org.gradle.testkit.runner.BuildResult;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@RunWith(Parameterized.class)
+public class ProjectLabelIT {
+  @Rule
+  public final ITGradleRunner gradleRunner = new ITGradleRunner();
+
+  @Parameterized.Parameters(name = "k8sResource {0} configured project-label ")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(
+        new Object[] { "default" },
+        new Object[] { "custom" }
+    );
+  }
+
+  @Parameterized.Parameter
+  public String expectedDir;
+
+  @Test
+  public void k8sResource_whenRun_generatesK8sManifestsWithProjectLabels() throws IOException, ParseException {
+    // When
+    ITGradleRunner itGradleRunner = gradleRunner.withITProject("project-label");
+    if (expectedDir.equals("custom")) {
+      itGradleRunner.withArguments("-Pjkube.enricher.jkube-project-label.useProjectLabel=true",
+          "-Pjkube.enricher.jkube-project-label.provider=custom-provider",
+          "k8sResource", "--stacktrace");
+    } else {
+      itGradleRunner.withArguments("k8sResource", "--stacktrace");
+    }
```

Comment:
Creating a parameterized test with an if-else condition inside checking for the parameter is NOT OK.

Reply:
If there's common setup or assert logic you want to verify, then move that to a separate method and call it from the specific test.

However, in this case, what you probably want to provide is a different set of arguments for each parameter.

```
  @Parameterized.Parameter
  public String arguments;
```